worlds/ff4fe/itempool.py

```diff
@@ -0,0 +1,88 @@
+from typing import List, Tuple
+
+from . import items
+from .locations import LocationData, free_character_locations, earned_character_locations
+from ..AutoWorld import World
+
+
+def create_itempool(locations: List[LocationData], multiworld: World) -> Tuple[List[str], str]:
+    chosen_character = get_chosen_character(multiworld)
+    character_pool = create_character_pool(multiworld, chosen_character)
+    key_item_pool = create_key_item_pool(multiworld)
+    location_count = len(locations) - len(character_pool) - len(key_item_pool) - 33  # Objective Status locations hack
+    if multiworld.is_vanilla_game():
+        location_count -= 1
+    if (multiworld.options.HeroChallenge.current_key != "none"
+            and multiworld.options.ForgeTheCrystal.current_key == "false"):
+        location_count -= 1  # We're manually placing the Advance Weapon at Kokkol
+    if multiworld.options.ConquerTheGiant.current_key == "true":
+        location_count -= 1  # No Kain3 location in Giant%
+    useful_percentage = multiworld.options.UsefulPercentage.value
+    useful_count = location_count * useful_percentage // 100
+    result_pool = []
+    result_pool.extend(character_pool)
+    result_pool.extend(key_item_pool)
+    result_pool.extend(multiworld.random.choices([item.name for item in items.useful_items
+                                                  if item.tier <= multiworld.options.MaxTier.value],
+                                                 k=useful_count))
+    result_pool.extend(multiworld.random.choices([item.name for item in items.filler_items
+                                                  if item.tier >= multiworld.options.MinTier.value],
+                                                 k=location_count - useful_count))
+    return (result_pool, chosen_character)
+
+
+def create_character_pool(multiworld: World, chosen_character: str) -> List[str]:
```

Comment:
```suggestion
def create_character_pool(world: World, chosen_character: str) -> List[str]:
```
okay I haven't seen this flipped the other way before. Would recommend doing world: World instead of multiworld: World

Reply:
and that's why it's a draft and I wanted eyes on it early because I knew I'd skim over half the stuff you mentioned at bare minimum. thank you, sincerely!